xbmc/pvr/PVRDatabase.cpp

```diff
@@ -436,6 +449,117 @@ int CPVRDatabase::GetPriority(const CPVRClient& client) const
   return atoi(strValue.c_str());
 }
 
+void CPVRDatabase::FixupClientIDs()
+{
+  // Get enabled and disabled PVR client addon infos
+  std::vector<ADDON::AddonInfoPtr> addonInfos;
+  CServiceBroker::GetAddonMgr().GetAddonInfos(addonInfos, false, ADDON::AddonType::PVRDLL);
+
+  std::vector<std::tuple<std::string, ADDON::AddonInstanceId, std::string>> clientInfos;
+  for (const auto& addonInfo : addonInfos)
+  {
+    const std::vector<ADDON::AddonInstanceId> instanceIds{addonInfo->GetKnownInstanceIds()};
+    for (const auto instanceId : instanceIds)
+    {
+      clientInfos.emplace_back(addonInfo->ID(), instanceId, addonInfo->Name());
+    }
+  }
+
+  for (const auto& clientInfo : clientInfos)
+  {
+    const std::string addonID{std::get<0>(clientInfo)};
```

Comment:
You could use structured bindings in the for loop to avoid this.
```c++
for (const auto& [addonID, instanceID, addonName] : clientInfos)
```

Reply:
Yep, thanks.